pandaserver/dataservice/adder_atlas_plugin.py

```diff
@@ -237,324 +235,321 @@ def update_outputs(self):
+                        elif (
+                            DataServiceUtils.getDestinationSE(file.destinationDBlockToken)
+                            and src_site_spec.ddm_output[scope_src_site_spec_output] == destination_se_site_spec.ddm_output[scope_dst_se_site_spec_output]
+                        ):
+                            tmp_dest_list = [DataServiceUtils.getDestinationSE(file.destinationDBlockToken)]
+                            # RSE is specified
+                        elif DataServiceUtils.getDistributedDestination(file.destinationDBlockToken):
+                            tmp_dest_list = [DataServiceUtils.getDistributedDestination(file.destinationDBlockToken)]
+                            dist_datasets.add(file_destination_dispatch_block)
+                            # RSE is specified for distributed datasets
+                        elif (
+                            src_site_spec.cloud != self.job.cloud
+                            and (not src_site_spec.ddm_output[scope_src_site_spec_output].endswith("PRODDISK"))
+                            and (self.job.prodSourceLabel not in ["user", "panda"])
+                        ):
+                            # T1 used as T2
+                            tmp_dest_list = [src_site_spec.ddm_output[scope_src_site_spec_output]]
+                        else:
+                            tmp_dest_list = []
+                            tmp_se_tokens = src_site_spec.setokens_output[scope_src_site_spec_output]
+                            for tmp_dest_token in file.destinationDBlockToken.split(","):
+                                if tmp_dest_token in tmp_se_tokens:
+                                    tmp_dest = tmp_se_tokens[tmp_dest_token]
+                                else:
+                                    tmp_dest = src_site_spec.ddm_output[scope_src_site_spec_output]
+                                if tmp_dest not in tmp_dest_list:
+                                    tmp_dest_list.append(tmp_dest)
+                        # add
+                        dataset_destination_map[file_destination_dispatch_block] = tmp_dest_list
+
+                # extra meta data
+                if self.ddm_backend == "rucio":
+                    if file.lfn in self.extra_info["lbnr"]:
+                        file_attrs["lumiblocknr"] = self.extra_info["lbnr"][file.lfn]
+                    if file.lfn in self.extra_info["nevents"]:
+                        file_attrs["events"] = self.extra_info["nevents"][file.lfn]
+                    elif self.extra_info["nevents"] != {}:
+                        file_attrs["events"] = None
+                    # if file.jediTaskID not in [0,None,'NULL']:
+                    #    fileAttrs['task_id'] = file.jediTaskID
+                    file_attrs["panda_id"] = file.PandaID
+                    if campaign not in ["", None]:
+                        file_attrs["campaign"] = campaign
+
+                # extract OS files
```

Comment:
What's an OS file? OS=Object Store? 10 years ago I think there was some Event Service development with Object Stores, but I think the experience with OS was not satisfactory and was broken off. Is this related? Is it still in use?

Reply:
This was originally implemented for files uploaded to object stores, but this code block is still used for files uploaded to alternative RSEs.